src/main/java/org/opensearch/neuralsearch/query/HybridQueryContext.java

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.query;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+
+@AllArgsConstructor
```

Comment:
We need to use either constructor or builder, but not both. You need to either remove constructor annotation or make it private. 
To enforce setting of required `pagination` param use NonNull annotation